Backend/src/Services/AuthenticationService.cs

```diff
@@ -156,19 +171,26 @@ public static void AddBearerClaimToResponse(MsalUiRequiredException ex, HttpResp
                     throw new AuthenticationException("SubjectAndAppToken is missing subject token");
                 }
 
-                return new AuthorizationContext { TenantObjectId = subjectTenantId };
+                if (requireTenantIdHeader)
+                {
+                    return new AuthorizationContext { TenantObjectId = subjectTenantId.Value };
+                }
+                else
+                {
+                    return new AuthorizationContext { };
+                }
             }
 
             var subjectClaims = ValidateAadTokenCommon(subjectAndAppToken.SubjectToken, openIdConnectConfiguration, isAppOnly: false);
             ValidateClaimValue(subjectClaims, "appid", appTokenAppId, "subject and app tokens should belong to same application");
-            ValidateClaimValue(subjectClaims, "tid", subjectTenantId.ToString(), "subject tokens must belong to the subject's tenant");
+            ValidateClaimValue(subjectClaims, "tid", subjectTenantId.Value.ToString(), "subject tokens must belong to the subject's tenant");
```

Comment:
Shouldn't we validate if subjectTenantId has value first? (In the case of subjectTenantId is null and requireTenantIdHeader is false)

Reply:
There is a check at line 154 - if (requireTenantIdHeader && !subjectTenantId.HasValue)